### PR TITLE
  #162414171 Email notification of new followers of a user and comments and response to an article.

### PR DESCRIPTION
**Pull request by ibraheemkabir**
**What does this PR do?**
This PR implements the feature which enables an author to receive email notification when a comment is made is made on his/her article.
This PR also enables a user to receive email notification when a comment is made on his.her response to an article i.e thread comments.

**Description of Task to be completed?**
Sends email notifications to users who have subscribed for email notifications
Adds email notification using send grid and HTML email templates.
Enables author's to receive email notification of comments on their articles
Enable users to receive email notifications of response to comments they made in an article.

**How should this be manually tested?**
To test these features:
**Email notification of a comment on user's articles**
Clone this repository and pull this branch.
Run npm `start`
create a user at the endpoint `api/v1/signup`
After receiving a token, create a new user with different details at the same endpoint.
Follow the first user using the second user's account at the endpoint `api/v1/users/'first-user-id'/follow`.
Create an article with the first user's account at the endpoint `api/v1/articles`
Check the second user's email, he/she should have received a notification of that article.
Make a comment on the first user's article at the endpoint `api/v1/articles/article-id/comment`.
Check the first user's email address, you should have received an email notification of that comment.

**Email notification of a thread comment on user's comment**
Clone this repository and pull this branch.
Run npm `start`
create a user at the endpoint `api/v1/signup`
After receiving a token, create a new user with different details at the same endpoint.
Follow the first user using the second user's account at the endpoint `api/v1/users/'first-user-id'/follow`.
Create an article with the first user's account at the endpoint `api/v1/articles`
Check the second user's email, he/she should have received a notification of that article.
Make a comment on the first user's article at the endpoint `api/v1/articles/article-id/comment`.
Check the first user's email address, you should have received an email notification of that comment.
Using the first user's account, respond to the comment made by the second user.
You should receive a notification of the comment on the second user's email address.

**What are the relevant pivotal tracker stories?**
PT Story: #162414171

**Screenshots**

screen shot 2019-02-02 at 1 21 41 pm
screen shot 2019-02-02 at 1 21 47 pm
screen shot 2019-02-02 at 1 22 10 pm

### CODE REVIEW

**Review by Luckzman — DISMISSED** [review]
LGTM

File: controllers/articlesCommentController.js [hunk 1]
```diff
@@ -14,17 +14,23 @@ const Comment = require('../lib/modelManagers/articlesComment'),
...
-    const { comment, userId, mainCommentId } = req.body;
+    const userId = req.decodedToken.payLoad;
+    const { comment, mainCommentId } = req.body;
     const { id } = req.params;
     const comments = await Comment.createComment(comment, userId, id, mainCommentId);
-    return successResponse(res, COMMENT_CREATED, RESOURCE_CREATED_CODE, comments);
+    const { articleId } = comments.dataValues;
+    req = {
+      id, comments, mainCommentId, userId, articleId
+    };
```
Comment (codrex) [inline controllers/articlesCommentController.js:33]:
```
req = {
      id, comments, mainCommentId, userId, articleId
    };
```
Why are you replacing req object with a new object?

File: lib/utils/sendNotifications.js [hunk 2]
```diff
@@ -33,6 +39,120 @@ const sendArticleNotificationToAuthorFollowers = async (id, title) => {
 };
 
 
+const sendNotificationToFollowedUser = async (id, followeeId) => {
+  const UserToNotify = await User.getUser(id);
+  const followerDetails = await User.getUser(followeeId);
+  if (UserToNotify.dataValues.userprofile.dataValues.emailnotification === true) {
```
Comment (codrex) [inline lib/utils/sendNotifications.js]:
```suggestion
  if (UserToNotify.userprofile.emailnotification === true) {
```
You don't have to use `dateValues` you can perform a direct property lookup on the instance object.

Comment (ibraheemkabir) [reply lib/utils/sendNotifications.js]: I have fixed this rex...

File: controllers/articlesCommentController.js (hunk 1 above)
Comment (ibraheemkabir) [reply controllers/articlesCommentController.js:33]: I was trying to pass the comment details as an object to the next middleware function, i have refactored this to req.commentDetails

**Review by codrex — APPROVED** [review]

**Pull request merged**